### PR TITLE
Add ospf_interfaces resource module

### PR DESCRIPTION
**Pull request by GomathiselviS**
Signed-off-by: GomathiselviS <gomathiselvi@gmail.com>

##### SUMMARY

Added new resource module vyos_ospf_interfaces.

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

File: changelogs/fragments/vyos_ospf_interfaces_rm.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+major_changes:
```
Comment (NilashishC) [inline changelogs/fragments/vyos_ospf_interfaces_rm.yaml:2]:
```suggestion
minor_changes:
```

File: tests/integration/targets/vyos_ospf_interfaces/tests/cli/replaced.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    - become: true
+      vyos.vyos.vyos_facts:
+        gather_network_resources: ospf_interfaces
+
+    - assert:
+        that:
+          - result.commands|length == 8
+          - result.changed == true
+          - result.commands|symmetric_difference(replaced.commands) == []
+          - result.after|symmetric_difference(ansible_facts['network_resources']['ospf_interfaces']) == []
```
Comment (NilashishC) [inline tests/integration/targets/vyos_ospf_interfaces/tests/cli/replaced.yaml:46]: Asserting before/after from gathered facts is not much useful, imo. Both are generated from the same code path.

File: plugins/module_utils/network/vyos/facts/ospf_interfaces/ospf_interfaces.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+
+class Ospf_interfacesFacts(object):
+    """The vyos ospf_interfaces facts class"""
+
+    def __init__(self, module, subspec="config", options="options"):
+        self._module = module
+        self.argument_spec = Ospf_interfacesArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
```
Comment (NilashishC) [inline plugins/module_utils/network/vyos/facts/ospf_interfaces/ospf_interfaces.py:38]: I think this section of code (37-46) is not required. I will remove it from builder template. But for now, removing it from here will bump code cov too.

**Review by NilashishC — CHANGES_REQUESTED** [review]

File: plugins/module_utils/network/vyos/facts/ospf_interfaces/ospf_interfaces.py (hunk 3 above)
Comment (GomathiselviS) [reply plugins/module_utils/network/vyos/facts/ospf_interfaces/ospf_interfaces.py:38]: I might have to change this assertion for most of the interfaces related tests. I shall take it up as a separate PR.

Comment (NilashishC) [reply plugins/module_utils/network/vyos/facts/ospf_interfaces/ospf_interfaces.py]: Okay

**Review by NilashishC — APPROVED** [review]

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Pull request merged**